app/javascript/react/src/containers/BathroomForm.js

```diff
@@ -0,0 +1,171 @@
+            address: this.state.address,
+            city: this.state.city,
+            state: this.state.state,
+            zip: this.state.zip,
+            gender: this.state.gender,
+            key_needed: this.state.keyNeeded,
+            toilet_quantity: parseInt(this.state.toiletQuantity, 10)
+          }
+        })
+      })
+      this.clearForms();
+    }
+  }
+
+  clearForms() {
+    this.setState({
+      establishment: '',
+      address: '',
+      city: '',
+      state: '',
+      zip: '',
+      gender: '',
+      keyNeeded: false,
+      toiletQuantity: 1
+    })
+  }
+
+  handleKeyNeededChange(event) {
+    if( this.state.keyNeeded === false ) {
+      this.setState({ keyNeeded: true });
+    } else {
+      this.setState({ keyNeeded: false });
+    }
+  }
+
+  validateInputs(value, name, label) {
+    if (value === '' || value === ' ') {
+      let newError = { [name]: `${label} can't be blank` };
+      this.setState({ errors: Object.assign(this.state.errors, newError) });
+      return false;
+    } else {
+      let errorState = this.state.errors;
+      delete errorState[name];
+      this.setState({ errors: errorState });
+      return true;
+    }
+  }
+
+  render() {
+    let errorDiv;
+    let errorItems;
+    if(Object.keys(this.state.errors).length > 0) {
+      errorItems = Object.values(this.state.errors).map(error => {
+        return( <p key={error}>{error}</p> )
+      })
+      errorDiv = <div className='callout alert'>{errorItems}</div>
+    }
+
+    return(
+      <form id='add-bathroom-form'>
+        <h3>Add New Bathroom:</h3>
+        {errorDiv}
+        <FormField
+          type='text'
+          name='establishment'
+          label='Establishment Name'
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.establishment}
+        />
+        <FormField
+          type='text'
+          name='address'
+          label='Address'
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.address}
+        />
+        <FormField
+          type='text'
+          name='city'
+          label='City'
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.city}
+        />
+        <FormField
+          type='text'
+          name='state'
+          label='State'
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.state}
+        />
+        <FormField
+          type='text'
+          name='zip'
+          label='Zip'
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.zip}
+        />
+        <FormField
+          type='text'
+          name='gender'
+          label='Gender'
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.gender}
+        />
+        <QuantitySelector
+          name="toiletQuantity"
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.toiletQuantity}
+        />
```

Comment:
Another way to cut down on repetition would be to make a hash and loop through the values to return one FormField each time

Reply:
Noted! We are planning to refactor this next week